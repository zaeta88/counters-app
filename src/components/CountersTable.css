.react-confirm-alert-body {
  font-family: "Inconsolata", monosapce;
  color: #000;
}

.actions .member-link {
  padding: 0 5px;
  border-radius: 3px;
  display: inline-block;
  cursor: pointer;
  padding: 4px  8px;
  margin: 0 4px;
  background-color: transparent;
  text-decoration: none;
}

.actions .member-link.disabled, .actions .member-link:disabled {
  opacity: 0.6;
  cursor: no-drop;
}
.actions .member-link:hover {
  background-color: #e1fce5;
}

.actions .member-link:disabled {
  background-color: gray;
}

.actions .member-link img {
  height: 22px;
}

.btn-secondary.dropdown-toggle, .react-confirm-alert-body > .react-confirm-alert-button-group button {
  color: #28a745;
  background-color: transparent;
  background-image: none;
  border: 1px solid #28a745;
  margin-right: 10px;
}

.btn-secondary:hover.dropdown-toggle, .btn-secondary:active.dropdown-toggle, .react-confirm-alert-body > .react-confirm-alert-button-group button:hover, .react-confirm-alert-body > .react-confirm-alert-button-group button:active {
  color: #fff;
  background-color: #28a745;
}

.carret img {
  width: 14px;
  transform: rotate(90deg);
}

.pagination, .right-button, .actions, .carret {
  float: right;
}

.page-item.active .page-link {
  background-color: #5bc66a;
  border-color: #5bc66a;
}

.page-link {
  color: #5bc66a;
}

.react-bootstrap-table-editing-cell {
  max-width: 30px;
}

.search-label, .item {
  margin-top: 5px;
  width: 100%;
}

.search-label input {
  background: url('../assets/img/search.svg') no-repeat 95% center;
  background-size: 20px 20px;
  padding-right: 30px;
  max-width: 220px;
  float: right;
  border-top: none;
  border-left: none;
  border-right: none;
  border-radius: 0;
}

.table-bordered tbody tr:first-child td {
    padding-top: .7rem;
}

.table-bordered {
  border: none;
}

.table-bordered td, .table-bordered th {
  border: none;
  padding: .3rem .75rem;
}

.table-bordered thead th {
  border-bottom: 2px solid #5bc66a;
}

.table-bordered thead tr th:last-child, .table-bordered thead tr th:nth-child(4) {
  width: 50px;
}

thead tr {
  border: 1px solid #edf3eb;
  background-color: #edf3eb;
  box-shadow: 0px 6px 8px -7px #798189;
}

.content-wrapper {
  padding: 10px 24px;
}

.content-wrapper hr {
  display: none;
}

.item img {
  height: 18px;
}