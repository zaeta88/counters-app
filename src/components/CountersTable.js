import React, { Component } from 'react';
import BootstrapTable from 'react-bootstrap-table-next';
import ToolkitProvider, { Search } from 'react-bootstrap-table2-toolkit';
import paginationFactory from 'react-bootstrap-table2-paginator';
import { confirmAlert } from 'react-confirm-alert';
import DeleteAlert from './DeleteAlert';
import CountersFilter from './CountersFilter';
import left from '../assets/img/arrow-left.svg';
import right from '../assets/img/arrow-right.svg';
import trash from '../assets/img/trash.svg';
import like from '../assets/img/like.svg';
import dislike from '../assets/img/dislike.svg';
import './CountersTable.css';
import 'react-confirm-alert/src/react-confirm-alert.css';

class CountersTable extends Component {
  constructor(props) {
    super(props);
    this.state = {
      counters: this.props.counters
    }
    this.countersLessThan = this.countersLessThan.bind(this);
    this.countersGreaterThan = this.countersGreaterThan.bind(this);
  }

  sortCaret = (order, column) => {
    if (!order) return (<span className="carret"><img src={left}/><img src={right}/></span>);
    else if (order === 'asc') return (<span className="carret"><img src={left}/></span>);
    else if (order === 'desc') return (<span className="carret"><img src={right}/></span>);
    return null;
  }

  customTotal = (from, to, size) => (
    <span className="react-bootstrap-table-pagination-total">
      Showing { from } to { to } of { size } Results
    </span>
  );

  tableColumns = () => {
    const columns = [{
      dataField: 'id',
      text: 'ID',
      sort: true,
      sortCaret: this.sortCaret
    }, {
      dataField: 'title',
      text: 'Title',
      sort: true,
      sortCaret: this.sortCaret
    }, {
      dataField: 'count',
      text: 'Count',
      sort: true,
      sortCaret: this.sortCaret,
      footer: '',
      footerFormatter: this.sumFormatter
    },  {
      dataField: '',
      text: 'Inc',
      align: 'center',
      formatter: this.plusButtonFormatter,
      events: {
        onClick: (e, column, columnIndex, row, rowIndex) => {
          this.props.handleIncrease(row)
        }
      }
    },  {
      dataField: '',
      text: 'Dec',
      align: 'center',
      formatter: this.lessButtonFormatter,
      events: {
        onClick: (e, column, columnIndex, row, rowIndex) => {
          this.props.handleDecrease(row)
        }
      }
    }, {
      dataField: '',
      text: 'Delete',
      align: 'center',
      formatter: this.deleteButtonFormatter,
      events: {
        onClick: (e, column, columnIndex, row, rowIndex) => {
          confirmAlert({
            customUI: ({ onClose }) => {
              return (
                <DeleteAlert 
                  handleDelete = { this.props.handleDelete }
                  onClose = { onClose }
                  row = { row }
                />
              );
            }
          });
        }
      }
    }];
    return columns;
  }

  tablePaginationOptions = (counters) => {
    const options = {
      paginationSize: 10,
      pageStartIndex: 1,
      showTotal: true,
      paginationTotalRenderer: this.customTotal,
      sizePerPageList: [{
        text: '10', value: 10
      }, {
        text: '25', value: 25
      }, {
        text: 'All', value: counters.length
      }] 
    };
    return options;
  }

  plusButtonFormatter = (cell, row) => {
    return (
      <div className="actions mb-0 like-dislike">
        <button className="member-link btn like" title="Increase count">
          <img src={like}/>
        </button>
      </div>
    );
  }

  lessButtonFormatter = (cell, row) => {
    return (
      <div className="actions mb-0 like-dislike">
        <button className="member-link btn dislike" title="Decrease count">
          <img src={dislike}/>
        </button>
      </div>
    );
  }
  
  deleteButtonFormatter = (cell, row) => {
    return (
      <div className="actions mb-0">
        <button className="member-link btn delete-btn" title="Delete">
          <img src={trash}></img>
        </button>
      </div>
    );
  }

  sumFormatter = (column, colIndex) => {
    const { counters } = this.state;
    const value = counters.length > 0 ? counters.map(o => o.count).reduce((a, b) => parseInt(a) + parseInt(b)) : 0;
    return (
      <h5><strong>Counters Total: { value }</strong></h5>
    );
  }

  countersLessThan = (value) => {
    let counters = value === '' ? this.props.counters : this.state.counters.filter((o) => parseInt(o.count) < parseInt(value));
    this.setState({ counters: counters })
  }

  countersGreaterThan = (value) => {
    let counters = value === '' ? this.props.counters : this.state.counters.filter((o) => parseInt(o.count) > parseInt(value));
    this.setState({ counters: counters });
  }

  emptyFilters = () => {
    this.setState({ counters: [] });
  }

  render() {
    const { SearchBar } = Search;
    const { counters } =  this.state;
    
    return (
      <div className="content-wrapper container">
        <ToolkitProvider
          keyField="id"
          data={ counters }
          columns={ this.tableColumns() }
          search
        >
          {
            props => (
              <div>
                <CountersFilter 
                  countersLessThan = { this.countersLessThan }
                  countersGreaterThan = { this.countersGreaterThan }
                  emptyFilters = { this.emptyFilters }
                />
                <SearchBar { ...props.searchProps } />
                <BootstrapTable
                  { ...props.baseProps }
                  noDataIndication="Empty Table"
                  pagination={ paginationFactory(this.tablePaginationOptions(counters)) }
                  hover
                />
              </div>
            )
          }
        </ToolkitProvider>
      </div>
    );
  }
}

export default CountersTable;