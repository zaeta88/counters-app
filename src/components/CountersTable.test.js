import React from 'react';
import { shallow } from 'enzyme';
import CountersTable from './CountersTable';


describe('React.Component CountersTable', () => {
  it('should render correctly', () => {
    const component = shallow(<CountersTable />);
    expect(component).toMatchSnapshot();
  });
});
