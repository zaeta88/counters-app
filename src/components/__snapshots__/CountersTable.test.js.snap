// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React.Component CountersTable should render correctly 1`] = `ShallowWrapper {}`;
